src/js/tracks/text-track.js

```diff
@@ -420,6 +421,16 @@ class TextTrack extends Track {
     this.cues.setCues_(this.cues_);
   }
 
+  /**
+   * Makes the text track serializable by removing circular dependencies.
+   *
+   * @return {Object} The track information as a serializable object
+   */
+  toJSON() {
```

Comment:
Should we be calling this `toJSON` when it's returning an object rather than a JSON string?

Noting we have this here too:
https://github.com/videojs/video.js/blob/8842d376ed3d834ad3ce0a5e662fe446513ea0bd/src/js/player.js#L4884-L4891

Reply:
it should be fine. From what i see from native behavior, the `toJSON()` method returns a serializable object rather than strictly a json string.